packages/astro/src/core/middleware/sequence.ts

```diff
@@ -0,0 +1,37 @@
+import type { APIContext, MiddlewareResponseHandler } from '../../@types/astro';
+import { defineMiddleware } from './index.js';
+import { AstroError, AstroErrorData } from '../errors/index.js';
+
+/**
+ * From SvelteKit: https://github.com/sveltejs/kit/blob/master/packages/kit/src/exports/hooks/sequence.js
+ *
+ * It accepts one or more middleware handlers and makes sure that they are run in sequence.
+ */
+export function sequence(...handlers: MiddlewareResponseHandler[]): MiddlewareResponseHandler {
```

Comment:
This JSDoc will be exposed to users, are we fine linking it to SvelteKit? 😅

Reply:
I agree that giving credits is good too, but maybe at the bottom of the jsdoc or at the astro docs, otherwise it gives the vibe of "we re-implement SvelteKit's feature, go check out their docs".